bodhi/server/util.py

```diff
@@ -1453,15 +1453,18 @@ def update_install_command(context, update):
         ValueError: When provided update is not in stable or testing state.
     """
     status = str(update.status)
-    alias = update.alias
     update_type = str(update.type)
+    testing_repo = 'updates-testing' if update.release.id_prefix == 'FEDORA' else 'epel-testing'
 
     if status != 'stable' and status != 'testing':
         raise ValueError('Only updates in stable or testing can be installed!')
 
-    command = 'sudo dnf {}{} --advisory={}{}'.format(
+    command = 'sudo {} {}{} --advisory={}{}'.format(
+        'dnf' if (update.release.id_prefix == 'FEDORA'
+                  and update.release.version_int >= 22) else 'yum',
         'install' if update_type == 'newpackage' else 'upgrade',
-        ' --enablerepo=updates-testing' if status == 'testing' else '',
-        alias,
+        (' --enablerepo=' + testing_repo) if status == 'testing' else '',
+        update.alias,
```

Comment:
I think a more general command would be useful, and I like your suggestion of adding a string to the Release model. Let's try that route instead. Sound good to you?

Reply:
I can think two way of doing this.
In the first one I can add a string and an Enum to the Release model. The string will carry the testing repository name, while the Enum will permit the choice of the package manager used by that release (currently 'yum' or 'dnf').

The second way is to create a new PackageManager model to handle all package managers stuff and reference it in the Release model.
For example:
```
class PackageManager(Base):
    """
    This model represent a package manager supported by Bodhi.

    In command and options strings the following placeholders are available:
    - {update}: the update alias
    - {options}: command options, like option_enable_testing
    - {testing_repository}: the testing repository name associated to the release

    Attributes:
        name (unicode): The name of the Package Manager.
        install_command (unicode): The string used to install a 'newpackage' update.
        upgrade_command (unicode): The string used to install the update.
        option_enable_testing (unicode): The string used to enable the testing repository.
    """

    __tablename__ = 'package_managers'
    __get_by__ = ('name',)
    __exclude_columns__ = ('id',)

    name = Column(UnicodeText, unique=True, nullable=False)
    install_command = Column(UnicodeText, nullable=False,
                             default='sudo dnf install {options} --advisory={update} \*')
    upgrade_command = Column(UnicodeText, nullable=False,
                             default='sudo dnf upgrade {options} --advisory={update}')
    option_enable_testing = Column(UnicodeText, nullable=False,
                                   default='--enablerepo={testing_repository}')

    release = relationship('Release', backref='package_managers', lazy=False)
```
and add in the Release model
```
package_manager_id = Column(Integer, ForeignKey('package_managers.id'))
testing_repository = Column(UnicodeText, nullable=False)
```

What would you prefer?